crates/ruma-macros/src/util.rs

```diff
@@ -221,3 +244,106 @@ fn is_serde_flatten_attribute(attr: &Attribute) -> bool {
 
     contains_flatten
 }
+
+/// Helper trait for a [`syn::Type`].
+pub(crate) trait TypeExt {
+    /// Get the inner type if this is wrapped in an `Option`.
+    fn option_inner_type(&self) -> Option<&syn::Type>;
+
+    /// Whether this type has a lifetime.
+    fn has_lifetime(&self) -> bool;
+}
+
+impl TypeExt for syn::Type {
+    fn option_inner_type(&self) -> Option<&syn::Type> {
+        let syn::Type::Path(syn::TypePath { path: syn::Path { segments, .. }, .. }) = self else {
+            return None;
+        };
+
+        if segments.last().unwrap().ident != "Option" {
+            return None;
+        }
+
+        let syn::PathArguments::AngleBracketed(syn::AngleBracketedGenericArguments {
```

Comment:
This is even worse x)
(just looks ugly, semantics are clear so I don't mind it staying this way)